### ISSUE TITLE
Progress Bar divide by zero error

### ISSUE THREAD

**Issue by bmabey** [jtriley/StarCluster#12]
Hi,
I've been using the latest version of StarCluster from github (for the add/remove node commands).  I've ran into the following error several times when starting a cluster.  Based on the stacktrace the error appears to be in the new progress bar code and it not handling the case where the denominator may be 0 initially.

```
PID: 39151 config.py:490 - DEBUG - Loading config
PID: 39151 config.py:107 - DEBUG - Loading file: /Users/bmabey/.starcluster/config
PID: 39151 config.py:490 - DEBUG - Loading config
PID: 39151 config.py:107 - DEBUG - Loading file: /Users/bmabey/.starcluster/config
PID: 39151 awsutils.py:54 - DEBUG - creating self._conn w/ connection_authenticator kwargs = {'path': '/', 'region': None, 'port': None, 'is_secure': True}
PID: 39151 start.py:167 - INFO - Using default cluster template: testcluster
PID: 39151 cluster.py:420 - DEBUG - plugin args = ['self', 'pkg_to_install']
PID: 39151 cluster.py:421 - DEBUG - plugin varargs = None
PID: 39151 cluster.py:422 - DEBUG - plugin keywords = None
PID: 39151 cluster.py:423 - DEBUG - plugin defaults = None
PID: 39151 cluster.py:440 - DEBUG - config_args = ['htop']
PID: 39151 cluster.py:441 - DEBUG - config_kwargs = {}
PID: 39151 ubuntu.py:7 - DEBUG - pkg_to_install = htop
PID: 39151 cluster.py:420 - DEBUG - plugin args = ['self']
PID: 39151 cluster.py:421 - DEBUG - plugin varargs = None
PID: 39151 cluster.py:422 - DEBUG - plugin keywords = None
PID: 39151 cluster.py:423 - DEBUG - plugin defaults = None
PID: 39151 cluster.py:440 - DEBUG - config_args = []
PID: 39151 cluster.py:441 - DEBUG - config_kwargs = {}
PID: 39151 cluster.py:1302 - INFO - Validating cluster template settings...
PID: 39151 cluster.py:1324 - INFO - Cluster template settings are valid
PID: 39151 cluster.py:1200 - INFO - Starting cluster...
PID: 39151 cluster.py:949 - INFO - Launching a 1-node cluster...
PID: 39151 cluster.py:892 - INFO - Launching master (ami: ami-a5c42dcc, type: m1.large)
PID: 39151 awsutils.py:175 - INFO - Creating security group @sc-testcluster...
PID: 39151 cluster.py:1023 - INFO - Waiting for cluster to come up... (updating every 30s)
PID: 39151 cluster.py:632 - DEBUG - existing nodes: {}
PID: 39151 cluster.py:649 - DEBUG - returning self._nodes = []
PID: 39151 cluster.py:1039 - INFO - Waiting for all nodes to be in a 'running' state...
PID: 39151 cli.py:171 - DEBUG - Traceback (most recent call last):
  File "build/bdist.macosx-10.6-universal/egg/starcluster/cli.py", line 152, in main
    sc.execute(args)
  File "build/bdist.macosx-10.6-universal/egg/starcluster/commands/start.py", line 195, in execute
    scluster.start(create=create, create_only=create_only, validate=False)
  File "build/bdist.macosx-10.6-universal/egg/starcluster/cluster.py", line 1191, in start
    return self._start(create, create_only)
  File "build/bdist.macosx-10.6-universal/egg/starcluster/utils.py", line 69, in wrap_f
    res = func(*arg, **kargs)
  File "build/bdist.macosx-10.6-universal/egg/starcluster/cluster.py", line 1209, in _start
    self._setup_cluster()
  File "build/bdist.macosx-10.6-universal/egg/starcluster/cluster.py", line 1223, in _setup_cluster
    self.wait_for_cluster()
  File "build/bdist.macosx-10.6-universal/egg/starcluster/cluster.py", line 1041, in wait_for_cluster
    pbar.update(0)
  File "build/bdist.macosx-10.6-universal/egg/starcluster/progressbar.py", line 312, in update
    self.prev_percentage = self.percentage()
  File "build/bdist.macosx-10.6-universal/egg/starcluster/progressbar.py", line 261, in percentage
    return self.currval * 100.0 / self.maxval
ZeroDivisionError: float division

PID: 39151 cli.py:173 - ERROR - Oops! Looks like you've found a bug in StarCluster
PID: 39151 cli.py:174 - ERROR - Debug file written to: /tmp/starcluster-debug-bmabey.log
PID: 39151 cli.py:175 - ERROR - Look for lines starting with PID: 39151
PID: 39151 cli.py:177 - ERROR - Please submit this file, minus any private information,
PID: 39151 cli.py:178 - ERROR - to starcluster@mit.edu
```

**Comment by jtriley** [comment]
Thanks for reporting, I'm working on this now. ~jtriley

**Comment by jtriley** [comment]
fix divide by 0 in wait_for_cluster progressbar

after waiting for spot requests, check that len(nodes) &gt; 0 before
waiting for nodes to become &#39;running&#39; in order to avoid a divide by
zero error when using the progressbar. this apparently happens because
StarCluster is some times too quick to query the nodes just after
requesting them. several users have reported this issue. this fix makes
sure that there&#39;s at least 1 node before attempting to display the
progress bar. the old spinner widget is used until len(nodes) &gt; 0.

moved spinner code into Cluster.get_spinner(). this function prints out
a custom log message, starts a spinner, and then returns the spinner object.

Closed by 21cd79f30c7c4c9bebd909802c5493cb7a56c9a8
